File: 백준/Bronze/8958. OX퀴즈/OX퀴즈.py
n = int(input())

data = []
for _ in range(n):
  data.append(list(input()))

result_data = []
for d in data:
  result = 0
  count = 0
  for i in d:
    if i == 'O':
      count += 1
      result += count
    else:
      count = 0
  result_data.append(result)

for i in result_data:
  print(i)

File: 백준/Bronze/8958. OX퀴즈/OX퀴즈.cc
#include <iostream>
#include <string>

using namespace std;

int main(void) {
	cin.tie(0)->sync_with_stdio(false);

	int n;
	cin >> n;

	
	for (int i = 0; i < n; i++) {
		int result = 0;
		int count = 0;

		string str;
		cin >> str;
		
		for (int j = 0; j < str.length(); j++) {
			if (str[j] == 'O') {
				result += ++count;
			 }
			else {
				count = 0;
			}
			
		}
		cout << result << '\n';
	}

	return 0;
}